lib/elasticity/index_config.rb

```diff
@@ -0,0 +1,50 @@
+module Elasticity
+  class IndexConfig
+    ATTRS = [:index_base_name, :document_type, :mapping, :strategy].freeze
```

Comment:
@divoxx some of the docnav clients are read only so they don't define a mapping, how would you feel about removing mapping as a required attribute?